### PR TITLE
Reduce network usage for messages

### PR DESCRIPTION
**Pull request by Astralcircle**
Less network load and can send more than 65532 characters in some cases (who needs it)

### CODE REVIEW

**Comment by Earu** [comment]
Hi, was this tested?

**Comment by Astralcircle** [comment]
Works fine for me and it just compresses the string so I don't think it will cause any problems

File: lua/easychat/networking.lua [hunk 1]
```diff
@@ -438,10 +439,10 @@ if CLIENT then
 	net.Receive(NET_BROADCAST_MSG, function()
 		local user_id = net.ReadUInt(16)
 		local user_name = net.ReadString()
-		local msg = net.ReadString()
 		local is_dead = net.ReadBool()
 		local is_team = net.ReadBool()
 		local is_local = net.ReadBool()
+		local msg = util.Decompress(net.ReadData(net.BytesLeft()))
```
[the remark this answers is not in the document]
Comment (Astralcircle) [reply lua/easychat/networking.lua:445]: Really? Can you provide an example of a message like this that doesn't decompress?

**Comment by Astralcircle** [comment]
I wouldn't say so. This message takes the most time and is always somewhere in the middle in terms of bits.
image

**Comment by Earu** [comment]
I mean at the end of the day it doesnt change anything functionally

File: lua/easychat/networking.lua (hunk 1 above)
[the remark this answers is not in the document]
Comment (Earu) [reply lua/easychat/networking.lua:445]: Sending the length can be abused the exact same way, its coming from the client anwyay

**Pull request merged**